bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -115,4 +129,70 @@
      */
     public void runNow(String ruleUID);
 
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a given scope or {@code null} scope.
+     * @param scope to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasScope(String scope) {
```

Comment:
Impl and should not be part of the interface. Also I'm not sure that the scope is characteristic of the rule. It is up to your business logic.